ironfish/src/wallet/wallet.ts

```diff
@@ -680,6 +681,28 @@ export class Accounts {
     try {
       this.assertHasAccount(sender)
 
+      // check if the chain data is fully synced
+      let headHash = this.headHashes.get(sender.id)
+      headHash = await sender.getHeadHash()
+      if (!headHash) {
+        headHash = this.headHashes.get(sender.id)
+      }
```

Comment:
`this.headHashes.get(sender.id)` and `sender.getHeadHash()` should give us the same value

I'm not sure we need the in-memory cache of head hashes, @NullSoldier ?